### DISCUSSION TITLE
Alternative to deprecated `separatedBy(separator, includeSeparators: false)`

### DISCUSSION THREAD

**Discussion by asartalo** [petitparser/dart-petitparser#161] [Q&A] [answered]
Hi again. Now that `separatedBy()` has been deprecated in favor of `plusSeparated()`, what is the current way of doing something like the following:

```dart
  Parser footer() =>
      ref0(footerLine).separatedBy(newLine, includeSeparators: false).plus();
```

Thanks!

**Comment by renggli** [comment, marked as answer]
The equivalent using `plusSeparated()` would be:

```dart
Parser footer() => ref0(footerLine)
     .plusSeparated(newLine)
     .map((list) => list.elements)     
     .plus();
```

The result of `plusSeparated` is a [SeparatedList](https://pub.dev/documentation/petitparser/latest/parser/SeparatedList-class.html) that contains the statically typed elements and separators (and some helpers). This is different to `separatedBy()` that had to return a single dynamically typed list.

**Comment by asartalo** [comment]
Thank you very much. Much appreciated!
